node/src/actors/json_rpc/json_rpc_methods.rs

```diff
@@ -534,13 +534,17 @@ pub fn get_block_chain(
 }
 
 /// Get block by hash
+///
+/// - First argument is the hash of the block that we are querying.
+/// - Second argument is whether we want the response to contain a list of hashes of the
+///   transactions found in the block.  
 /* test
-{"jsonrpc":"2.0","id":1,"method":"getBlock","params":["c0002c6b25615c0f71069f159dffddf8a0b3e529efb054402f0649e969715bdb"]}
-{"jsonrpc":"2.0","id":1,"method":"getBlock","params":[{"SHA256":[255,198,135,145,253,40,66,175,226,220,119,243,233,210,25,119,171,217,215,188,185,190,93,116,164,234,217,67,30,102,205,46]}]}
+{"jsonrpc":"2.0","id":1,"method":"getBlock","params":["c0002c6b25615c0f71069f159dffddf8a0b3e529efb054402f0649e969715bdb", false]}
+{"jsonrpc":"2.0","id":1,"method":"getBlock","params":[{"SHA256":[255,198,135,145,253,40,66,175,226,220,119,243,233,210,25,119,171,217,215,188,185,190,93,116,164,234,217,67,30,102,205,46, false]}]}
 */
-pub fn get_block(hash: Result<(Hash,), jsonrpc_core::Error>) -> JsonRpcResultAsync {
-    let hash = match hash {
-        Ok(x) => x.0,
+pub fn get_block(args: Result<(Hash, bool), jsonrpc_core::Error>) -> JsonRpcResultAsync {
```

Comment:
This is a breaking change, but there is a way to make it non-breaking and default the bool to true to keep the old behavior. We need to pass the args as a json value and do the parsing manually.

```suggestion
pub fn get_block(args: Value) -> JsonRpcResultAsync {
```

Reply:
Done :wink: